src/components/themr.js

```diff
@@ -134,16 +137,19 @@ export default (componentName, localTheme, options = {}) => (ThemedComponent) =>
 
     render() {
       let renderedElement
+      //exclude themr-only props
+      //noinspection JSUnusedLocalSymbols
+      const { composeTheme, themeNamespace, ...props } = this.props //eslint-disable-line no-unused-vars
 
       if (optionWithRef) {
         renderedElement = React.createElement(ThemedComponent, {
-          ...this.props,
+          ...props,
           ref: 'wrappedInstance',
           theme: this.theme_
```

Comment:
Almost everywhere it is extracted: `const {theme, ...props} = this.props; return <input {...props}/>`;